### PR TITLE
Add logic to preserve process.exitCode and set it to "1" on error

### PR DESCRIPTION
**Pull request by jrop**

### CODE REVIEW

**Comment by lukeed** [comment]
Any particular use case where this is useful?

**Comment by jrop** [comment]
Yes, we use an automated build-system (Jenkins), and if the exit code of any commands in the build-script is/are not zero, then Jenkins marks the build as failed.

When using gulp, the exit code is non-zero when the build fails as well.

File: cli.js [hunk 1]
```diff
@@ -8,4 +8,7 @@ var args = ['--harmony'].concat(fly).concat(process.argv.slice(2))
 /**
  * Spawn a new Node process
  */
-require('child_process').spawn(node, args, {stdio: 'inherit'})
+var proc = require('child_process').spawn(node, args, {stdio: 'inherit'})
```
Comment (ghost) [inline cli.js:11]: Are we going to use `proc` somewhere else? I think you could simply write:

```
require('child_process').spawn(node, args, {stdio: 'inherit'}).on('close', ...)
```

**Comment by ghost** [comment]
@jrop I left you one comment, other than that, LGTM.

**Comment by jrop** [comment]
@bucaran you are correct.  Would you like me to change it to the single line and recommit?

**Comment by lukeed** [comment]
@jrop he meant to remove the `proc` variable assignment :) 

Sorry haven't merged yet. Been thinking about this & there's a better way to do it, but this'll work for now.

**Comment by jrop** [comment]
@lukeed Yep I knew what he meant.  I'm reviewing these on my phone so my communication skills are probably on the less verbose side :)

**Comment by lukeed** [comment]
Haha can't blame ya

**Comment by lukeed** [comment]
I'll merge & make the changes

**Comment by ghost** [comment]
Thanks @lukeed

**Comment by pine** [comment]
Thank you :+1: @lukeed @jrop

**Pull request merged**